tensorboard/plugins/interactive_inference/witwidget/notebook/colab/wit.py

```diff
@@ -233,9 +236,14 @@ def set_examples(self, examples):
       self._generate_sprite()
 
   def infer(self):
-    inferences = base.WitWidgetBase.infer_impl(self)
-    output.eval_js("""inferenceCallback('{inferences}')""".format(
-      inferences=json.dumps(inferences)))
+    try:
+      inferences = base.WitWidgetBase.infer_impl(self)
+      output.eval_js("""inferenceCallback('{inferences}')""".format(
+        inferences=json.dumps(inferences)))
+    except Exception as e:
+      error_str = str(e)
+      output.eval_js("""backendError('{error}')""".format(
```

Comment:
Likewise an injection vector, and this one is may be plausibly hit even
in a non-malicious case, because Python exception texts frequently
contain single quotes:

```python
>>> import json
>>> try: None.wat
... except Exception as e:
...   js = """backendError('{error}')""".format(error=json.dumps(str(e)))
...   print(js)
... 
backendError('"'NoneType' object has no attribute 'wat'"')
>>> # ^ JS syntax error
```

(More occurrences below.)

Reply:
replaced with object and not string.